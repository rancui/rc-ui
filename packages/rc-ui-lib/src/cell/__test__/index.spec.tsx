import React from 'react';
import { cleanup, render } from '@testing-library/react';
import Cell from '..';

describe('Cell', () => {
  afterEach(() => {
    cleanup();
    jest.restoreAllMocks();
  });

  it('should render value slot correctly', () => {
    const { container } = render(<Cell value="Custom Value" />);
    expect(container).toMatchSnapshot();
  });

  it('should render title slot correctly', () => {
    const { container } = render(<Cell title="Custom Title" />);
    expect(container).toMatchSnapshot();
  });

  it('should render label slot correctly', () => {
    const { container } = render(<Cell title="Title" label="Custom Label" />);
    expect(container).toMatchSnapshot();
  });

  it('should render icon slot correctly', () => {
    const { container } = render(<Cell icon="location-o" />);
    expect(container).toMatchSnapshot();
  });

  it('should render extra slot correctly', () => {
    const { container } = render(<Cell title="Title" value="Value" extra="Custom Extra" />);
    expect(container).toMatchSnapshot();
  });

  it('should change arrow direction when using arrow-direction prop', () => {
    const { container } = render(<Cell title="Title" value="Value" isLink arrowDirection="down" />);
    expect(container.querySelector('.rc-cell__right-icon')).toBeTruthy();
    expect(container.querySelector('.van-icon-arrow-down')).toBeTruthy();
  });

  it('should change title style when using title-style prop', () => {
    const { container } = render(<Cell title="Title" titleStyle={{ color: 'red' }} />);
    const title = container.querySelector('.rc-cell__title');
    const style = getComputedStyle(title);
    expect(style.color).toEqual('red');
  });

  it('should change icon class prefix when using icon-prefix prop', () => {
    const { container } = render(<Cell title="Title" icon="success" iconPrefix="my-icon" />);
    expect(container).toMatchSnapshot();
  });

  it('should allow to disable clicakble when using is-link prop', () => {
    const { container } = render(<Cell title="Title" isLink clickable={false} />);
    expect(container.querySelector('.rc-cell--clickable')).toBeTruthy();
  });

  it('should render correctly when wrapperd with Cell.Group', () => {
    const { container } = render(
      <Cell.Group>
        <Cell title="单元格" value="内容" />
        <Cell title="单元格" value="内容" label="描述信息" />
      </Cell.Group>,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render correctly when using title prop in Cell.Group', () => {
    const { container } = render(
      <div>
        <Cell.Group title="分组1">
          <Cell title="单元格" value="内容" />
        </Cell.Group>
        <Cell.Group title="分组2">
          <Cell title="单元格" value="内容" />
        </Cell.Group>
      </div>,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render correctly when using inset prop in Cell.Group', () => {
    const { container } = render(
      <Cell.Group inset>
        <Cell title="单元格" value="内容" />
        <Cell title="单元格" value="内容" label="描述信息" />
      </Cell.Group>,
    );
    expect(container).toMatchSnapshot();
  });
});
