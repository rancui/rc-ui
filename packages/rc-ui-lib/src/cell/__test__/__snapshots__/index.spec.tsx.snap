// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell should change icon class prefix when using icon-prefix prop 1`] = `
<div>
  <div
    class="rc-cell"
  >
    <i
      class="rc-badge__wrapper rc-cell__left-icon my-icon my-icon-success"
    />
    <div
      class="rc-cell__title"
    >
      Title
    </div>
  </div>
</div>
`;

exports[`Cell should render correctly when using inset prop in Cell.Group 1`] = `
<div>
  <div
    class="rc-cell-group rc-cell-group--inset"
  >
    <div
      class="rc-cell"
    >
      <div
        class="rc-cell__title"
      >
        单元格
      </div>
      <div
        class="rc-cell__value"
      >
        <span>
          内容
        </span>
      </div>
    </div>
    <div
      class="rc-cell"
    >
      <div
        class="rc-cell__title"
      >
        单元格
        <div
          class="rc-cell__label"
        >
          描述信息
        </div>
      </div>
      <div
        class="rc-cell__value"
      >
        <span>
          内容
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Cell should render correctly when using title prop in Cell.Group 1`] = `
<div>
  <div>
    <div
      class="rc-cell-group__title"
    >
      分组1
    </div>
    <div
      class="rc-cell-group rc-hairline--top-bottom"
    >
      <div
        class="rc-cell"
      >
        <div
          class="rc-cell__title"
        >
          单元格
        </div>
        <div
          class="rc-cell__value"
        >
          <span>
            内容
          </span>
        </div>
      </div>
    </div>
    <div
      class="rc-cell-group__title"
    >
      分组2
    </div>
    <div
      class="rc-cell-group rc-hairline--top-bottom"
    >
      <div
        class="rc-cell"
      >
        <div
          class="rc-cell__title"
        >
          单元格
        </div>
        <div
          class="rc-cell__value"
        >
          <span>
            内容
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Cell should render correctly when wrapperd with Cell.Group 1`] = `
<div>
  <div
    class="rc-cell-group rc-hairline--top-bottom"
  >
    <div
      class="rc-cell"
    >
      <div
        class="rc-cell__title"
      >
        单元格
      </div>
      <div
        class="rc-cell__value"
      >
        <span>
          内容
        </span>
      </div>
    </div>
    <div
      class="rc-cell"
    >
      <div
        class="rc-cell__title"
      >
        单元格
        <div
          class="rc-cell__label"
        >
          描述信息
        </div>
      </div>
      <div
        class="rc-cell__value"
      >
        <span>
          内容
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Cell should render extra slot correctly 1`] = `
<div>
  <div
    class="rc-cell"
  >
    <div
      class="rc-cell__title"
    >
      Title
    </div>
    <div
      class="rc-cell__value"
    >
      <span>
        Value
      </span>
    </div>
    Custom Extra
  </div>
</div>
`;

exports[`Cell should render icon slot correctly 1`] = `
<div>
  <div
    class="rc-cell"
  >
    <i
      class="rc-badge__wrapper rc-cell__left-icon van-icon van-icon-location-o"
    />
  </div>
</div>
`;

exports[`Cell should render label slot correctly 1`] = `
<div>
  <div
    class="rc-cell"
  >
    <div
      class="rc-cell__title"
    >
      Title
      <div
        class="rc-cell__label"
      >
        Custom Label
      </div>
    </div>
  </div>
</div>
`;

exports[`Cell should render title slot correctly 1`] = `
<div>
  <div
    class="rc-cell"
  >
    <div
      class="rc-cell__title"
    >
      Custom Title
    </div>
  </div>
</div>
`;

exports[`Cell should render value slot correctly 1`] = `
<div>
  <div
    class="rc-cell"
  >
    <div
      class="rc-cell__value rc-cell__value--alone"
    >
      <span>
        Custom Value
      </span>
    </div>
  </div>
</div>
`;
